showcases/react-showcase/src/components/index.tsx

```diff
@@ -1,51 +1,62 @@
-import { type CSSProperties, type ReactElement } from 'react';
-import { DBDivider } from '../../../../output/react/src';
+import { DBCodeDocs, DBDivider, DBLink } from '../../../../output/react/src';
+import useQuery from '../hooks/use-query';
+import {
+	type DefaultComponentProps,
+	type DefaultComponentVariants
+} from './data';
 
-export type DefaultComponentExample = {
-	name: string;
-	example: ReactElement;
-	style?: CSSProperties;
-
-	className?: string;
-};
-
-export type DefaultComponentVariants = {
-	name: string;
-	examples: DefaultComponentExample[];
-};
-
-export type DefaultComponentProps = {
-	title: string;
-	description?: ReactElement;
-	variants: DefaultComponentVariants[];
-};
+const VariantList = ({ examples }: DefaultComponentVariants) => (
+	<DBCodeDocs
+		className="variants-card"
+		codeSnippets={examples
+			.filter((example) => example.code)
+			.map((example) => `/* ${example.name} */\n${example.code}`)}>
+		<ul className="variants-list">
+			{examples.map((example, exampleIndex) => (
+				<li
+					key={`${example.name}-${exampleIndex}`}
+					style={example.style}
+					className={example.className}>
+					<p className="example-name">
+						<small>{example.name}</small>
+					</p>
+					{example.example}
+				</li>
+			))}
+		</ul>
+	</DBCodeDocs>
+);
 
 const DefaultComponent = ({
 	title,
 	description,
 	variants
 }: DefaultComponentProps) => {
+	const pageName = useQuery()[4];
+	const setPageName = useQuery()[5];
+
+	if (pageName) {
+		const foundVariant = variants.find(
+			(variant) => variant.name.toLowerCase() === pageName
+		);
+		if (foundVariant) {
+			return <VariantList {...foundVariant} />;
+		}
+	}
+
 	return (
 		<div className="default-container">
 			<h1>{title}</h1>
 			{description ?? <></>}
 			{variants?.map((variant, index) => (
 				<div key={`${variant.name}-${index}`}>
 					<DBDivider></DBDivider>
-					<strong>{variant.name}</strong>
-					<ul className="variants-list">
-						{variant.examples.map((example, exampleIndex) => (
-							<li
-								key={`${example.name}-${exampleIndex}`}
-								style={example.style}
-								className={example.className}>
-								<p>
-									<small>{example.name}</small>
-								</p>
-								{example.example}
-							</li>
-						))}
-					</ul>
+					<DBLink
+						content="external"
+						onClick={() => setPageName(variant.name.toLowerCase())}>
+						<strong>{variant.name}</strong>
```

Comment:
```suggestion
						{variant.name}
```

This could better achieve via CSS.